### PR TITLE
Redo table columns for i18n in terraware-web

### PR DESCRIPTION
**Pull request by karthikbtf**
- wasn't too thrilled about the last fix
- extracted out changed-locale semantics and moved it into the table wrapper
- now each component supplies a function that returns the table names, function has to be outside scope of the component so it doesn't change on every render
- this will scale more easily without bugs, etc.

### CODE REVIEW

File: src/components/common/table/index.tsx [hunk 1]
```diff
@@ -109,23 +109,29 @@ export function OrderPreserveableTable<T extends TableRowType>(
...
-        columns.find((sourceColumn: TableColumnType) => sourceColumn.key === tableColumn.key)
+        sourceColumns.find((sourceColumn: TableColumnType) => sourceColumn.key === tableColumn.key)
       )
       .filter((tableColumn?: TableColumnType) => !!tableColumn) as TableColumnType[];
 
     if (!_.isEqual(tableColumns, refreshedColumns)) {
       setTableColumns(refreshedColumns);
     }
-  }, [columns, tableColumns]);
+  }, [activeLocale, columns, tableColumns]);
```
Comment (karthikbtf) [inline src/components/common/table/index.tsx:134]: now the table takes care of locale changes, only gotcha is the the columns callback needs to be in a global scope or within a useCallback (in the calling client)

**Review by charlie-tf — APPROVED** [review]
Change makes sense. I agree that centralizing this logic in the table makes more sense and will scale better.

**Comment by karthikbtf** [comment]
> Change makes sense. I agree that centralizing this logic in the table makes more sense and will scale better.

Thanks Charlie, this approach requires that we flag any `getColumns` that is transient within the scope of a component - should be a useCallback or declared outside the component. Hope we don't find that cumbersome. Downside to not doing this is a few extra renders of the table.

**Comment by charlie-tf** [comment]
> > Change makes sense. I agree that centralizing this logic in the table makes more sense and will scale better.
> 
> Thanks Charlie, this approach requires that we flag any `getColumns` that is transient within the scope of a component - should be a useCallback or declared outside the component. Hope we don't find that cumbersome. Downside to not doing this is a few extra renders of the table.

I think that's a reasonable requirement

**Comment by karthikbtf** [comment]
> > > Change makes sense. I agree that centralizing this logic in the table makes more sense and will scale better.
> > 
> > 
> > Thanks Charlie, this approach requires that we flag any `getColumns` that is transient within the scope of a component - should be a useCallback or declared outside the component. Hope we don't find that cumbersome. Downside to not doing this is a few extra renders of the table.
> 
> I think that's a reasonable requirement

cc @constanzauanini for context

**Pull request merged**